### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by vnijat93**
## Description

[Describe the changes that you've made in this pull request and explain why they are necessary.]

## Checklist

Please ensure that your pull request meets the following requirements:

- [ ] Code compiles without errors and warnings
- [ ] Code follows the project's coding style guidelines
- [ ] Code is tested and test cases pass
- [ ] Code has been reviewed by at least one other developer

## Related Issues

[If your pull request is related to any issues, link them here.]

## Additional Notes

[Include any additional information or notes that may be helpful in reviewing your pull request.]

### CODE REVIEW

File: src/FileManager.cpp [hunk 1]
```diff
@@ -127,8 +127,7 @@ array<string, 2> FileManager::readFile(string inputPath) {
       string data = "";
       string line = "";
       // while we are not at the end of the file
-      while (file) {
-        getline(file, line);
+      while (getline(file, line)) {
```
Comment (vnijat93) [inline src/FileManager.cpp:130]: we have agreed on removing this line

File: src/Map.cpp [hunk 2]
```diff
@@ -201,5 +201,5 @@ void Map::map(const string& aKey, const string& aValue) {
 // exports tokenized values to the disk
 void Map::exportData(const string& aKey, const string& aValue) {
     // write the data to the file
-    FileManager::writeFile(FileManager::MODE::APPEND, outputPath, aKey, aValue);
+    FileManager::writeFile(FileManager::MODE::APPEND, outputPath, aValue);
```
Comment (vnijat93) [inline src/Map.cpp:204]: the `aKey` is the input file name and it should not be used when writing temp file

File: src/Sort.cpp [hunk 3]
```diff
@@ -48,12 +48,12 @@ void Sort::Sorter() {
     }
 
     // Sort the word counts and write them to the output file
-    SortCountsToFile(word_counts);
+    SortCountsToFile(word_counts, output_filename_);
```
Comment (vnijat93) [inline src/Sort.cpp:51]: we need to write final output to a temp file

File: src/Sort.cpp [hunk 4]
```diff
@@ -48,12 +48,12 @@ void Sort::Sorter() {
...
 
 
     ///////////////////////////////////////////////////////////////////////////////
     // print to output_filename 
-    FileManager::writeFile(FileManager::CREATE, output_filename_, result);
+    // FileManager::writeFile(FileManager::CREATE, output_filename_, result);
```
Comment (vnijat93) [inline src/Sort.cpp:56]: this was not writing anything since the `result` is empty

File: src/Sort.cpp [hunk 5]
```diff
@@ -136,8 +136,10 @@ void Sort::SortCountsToFile(const map<string, vector<int>>& word_counts) {
         }
     }
 
+    FileManager::writeFile(FileManager::CREATE, outputFilePath, result);
```
Comment (vnijat93) [inline src/Sort.cpp:139]: writing result to a file

**Review by saie100 — APPROVED** [review]

**Pull request merged**